pkg/apis/crds/karpenter.k8s.aws_ec2nodeclasses.yaml

```diff
@@ -143,18 +155,41 @@ spec:
                             and sc1: 125-16,384 \n * standard: 1-1,024"
                           pattern: ^(\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))(([KMGTPE]i)|[numkMGTPE]|([eE](\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))))?$
                           x-kubernetes-int-or-string: true
+                          x-kubernetes-validations:
+                          - rule: self.endsWith('Gi') || self.endsWith('Ti') || self.endsWith('G')
```

Comment:
Would it be simpler to enforce this with pure regex? Your call but I think you might have an easier time writing the rule vs doing 4 or 5 rules here that do conversions into doubles. Regex should also be less computationally expensive  then the current ruleset that you have

Reply:
We can't use Regex on an object that is not a string

Reply:
Isn't this a string? Or does it fail because this is an int-or-string. I'd assume that this is being stored by the apiserver on the backend as a string. You could also override this to be a string type since I don't think we are going to support small values like integers anyways